packages/bundle-source/src/index.js

```diff
@@ -140,21 +140,29 @@ async function transformSource(
 
 async function bundleZipBase64(startFilename, dev, powers = {}) {
   const entry = url.pathToFileURL(path.resolve(startFilename));
-  const bytes = await makeArchive({ ...readPowers, ...powers }, entry, {
-    dev,
-    moduleTransforms: {
-      async mjs(sourceBytes) {
-        const source = textDecoder.decode(sourceBytes);
-        const { code: object } = await transformSource(source, {
-          sourceType: 'module',
-        });
-        const objectBytes = textEncoder.encode(object);
-        return { bytes: objectBytes, parser: 'mjs' };
+  const { bytes, sha512 } = await makeAndHashArchive(
+    { ...readPowers, ...powers },
+    entry,
+    {
+      dev,
+      moduleTransforms: {
+        async mjs(sourceBytes) {
+          const source = textDecoder.decode(sourceBytes);
+          const { code: object } = await transformSource(source, {
+            sourceType: 'module',
+          });
+          const objectBytes = textEncoder.encode(object);
+          return { bytes: objectBytes, parser: 'mjs' };
+        },
       },
     },
-  });
+  );
   const endoZipBase64 = encodeBase64(bytes);
-  return harden({ endoZipBase64, moduleFormat: 'endoZipBase64' });
+  return harden({
+    endoZipBase64,
+    endoZipBase64Sha512: sha512,
+    moduleFormat: 'endoZipBase64',
```

Comment:
nit: I'd put `moduleFormat` first, to emphasize that it's the first thing a reader needs to pay attention to ("don't use the presence of `.endoZipBase64` to discover the module format, we have a key to tell you that")